### PR TITLE
Added device uuid or, serial number to http messages, while communicating with zedcloud, for ease of debug.

### PR DESCRIPTION
**Pull request by srinibas-zededa**

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/handleconfig.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ func handleConfigInit() {
 		log.Fatal("uuid.FromString", err, string(b))
 	}
 	log.Infof("Read UUID %s\n", devUUID)
+	zedcloudCtx.DevUUIDStr = uuidStr
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handleconfig.go]: Better to use devUUID.String() here even if it is the same value; the uuidStr is an intermediate result as part of the parsing

File: pkg/pillar/zedcloud/send.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ type ZedCloudContext struct {
 	FailureFunc         func(intf string, url string, reqLen int64, respLen int64)
 	SuccessFunc         func(intf string, url string, reqLen int64, respLen int64)
 	NoLedManager        bool // Don't call UpdateLedManagerConfig
+	DevUUIDStr          string
```
Comment (eriknordmark) [inline pkg/pillar/zedcloud/send.go]: We should define this as a uuid.UUID type

File: pkg/pillar/zedcloud/send.go [hunk 3]
```diff
@@ -241,6 +243,13 @@ func SendOnIntf(ctx ZedCloudContext, destUrl string, intf string, reqlen int64,
 
 		if b != nil {
 			req.Header.Add("Content-Type", "application/x-proto-binary")
+			if ctx.DevUUIDStr != "" {
+				req.Header.Add("X-DEVICE-ID", ctx.DevUUIDStr)
+			} else {
+				if ctx.SerialNumStr != "" {
+					req.Header.Add("X-DEVICE-ID", ctx.SerialNumStr)
```
Comment (eriknordmark) [inline pkg/pillar/zedcloud/send.go]: The serial number isn't unique; we should combine this with (a hash of) the onboarding cert for it to be useful for tracing.

Comment (eriknordmark) [reply pkg/pillar/zedcloud/send.go]: And isn't it confusing to use the same X-DEVICE-ID header for two completely different things?
I'd suggest putting a UUID in X-Request-Id and handling the non-UUID (pre-onboarding) case differently.

Comment (srinibas-zededa) [reply pkg/pillar/zedcloud/send.go]: Have made the X-DEVICE-ID to be X-Request-Id. For pre-onboard/registration, the tracing field is TBD.

File: pkg/pillar/cmd/zedagent/handleconfig.go (hunk 1 above)
Comment (srinibas-zededa) [reply pkg/pillar/cmd/zedagent/handleconfig.go]: Sure

File: pkg/pillar/zedcloud/send.go (hunk 2 above)
Comment (srinibas-zededa) [reply pkg/pillar/zedcloud/send.go]: done

File: pkg/pillar/zedcloud/send.go (hunk 3 above)
Comment (eriknordmark) [reply pkg/pillar/zedcloud/send.go]: OK, but I realize you should add the headers even when b is nil; b is set for a POST but I think we want these additional headers for a GET request as well.

File: pkg/pillar/zedcloud/send.go [hunk 4]
```diff
@@ -241,6 +243,10 @@ func SendOnIntf(ctx ZedCloudContext, destUrl string, intf string, reqlen int64,
 
 		if b != nil {
 			req.Header.Add("Content-Type", "application/x-proto-binary")
+			devUuidStr := ctx.DevUUID.String()
```
Comment (eriknordmark) [inline pkg/pillar/zedcloud/send.go]: Move this added code outside of the if b != nil statement so it gets added for GET requests and not just POSTs

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**